tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <chrono>
+#include <concepts>
+#include <format>
+#include <iostream>
+#include <stdio.h>
+#include <string_view>
+#include <type_traits>
+#include <utility>
+
+using namespace std;
+using namespace chrono;
+
+#ifndef __clang__ // TRANSITION, LLVM-48606
+
+// copied from the string_view tests
+template <typename CharT>
+struct choose_literal; // not defined
+
+template <>
+struct choose_literal<char> {
+    static constexpr const char* choose(const char* s, const wchar_t*) {
+        return s;
+    }
+};
+
+template <>
+struct choose_literal<wchar_t> {
+    static constexpr const wchar_t* choose(const char*, const wchar_t* s) {
+        return s;
+    }
+};
+
+#define TYPED_LITERAL(CharT, Literal) (choose_literal<CharT>::choose(Literal, L##Literal))
+
+template <typename CharT>
+struct testing_callbacks {
+    _Align expected_alignment = _Align::_None;
+    basic_string_view<CharT> expected_fill;
+    int expected_width                   = -1;
+    size_t expected_dynamic_width        = static_cast<size_t>(-1);
+    bool expected_auto_dynamic_width     = false;
+    int expected_precision               = -1;
+    size_t expected_dynamic_precision    = static_cast<size_t>(-1);
+    bool expected_auto_dynamic_precision = false;
+    vector<_Chrono_specs<CharT>>& expected_chrono_specs;
+    size_t curr_index = 0;
+
+    constexpr void _On_align(_Align aln) {
+        assert(aln == expected_alignment);
+    }
+    constexpr void _On_fill(basic_string_view<CharT> str_view) {
+        assert(str_view == expected_fill);
+    }
+    constexpr void _On_width(int width) {
+        assert(width == expected_width);
+    }
+    constexpr void _On_dynamic_width(size_t id) {
+        assert(id == expected_dynamic_width);
+    }
+    constexpr void _On_dynamic_width(_Auto_id_tag) {
+        assert(expected_auto_dynamic_width);
+    }
+    constexpr void _On_precision(int pre) {
+        assert(pre == expected_precision);
+    }
+    constexpr void _On_dynamic_precision(size_t id) {
+        assert(id == expected_dynamic_precision);
+    }
+    constexpr void _On_dynamic_precision(_Auto_id_tag) {
+        assert(expected_auto_dynamic_precision);
+    }
+    constexpr void _On_conversion_spec(CharT mod, CharT type) {
+        char _Char_mod  = static_cast<char>(expected_chrono_specs[curr_index]._Modifier);
+        char _Char_type = static_cast<char>(expected_chrono_specs[curr_index]._Type);

```

Comment:
`_Modifier` and `_Type` are already `char` so the `static_cast<char>` is unnecessary:
```cpp
    template <class _CharT>
    struct _Chrono_specs {
        _CharT _Lit_char = _CharT{0}; // any char other than {, }, %
        char _Modifier   = '\0'; // either E or O
        char _Type       = '\0';
    };
```
Was the intention to cast `CharT mod` and `CharT type` to `char` to avoid sign conversion warnings (in which case the assertions below should be changed accordingly)?